tests/acceptance/features/bootstrap/NotificationContext.php

```diff
@@ -134,4 +142,102 @@ public function theJsonDataFromLastResponseShouldMatch(
 			$this->featureContext->getJSONSchema($schemaString)
 		);
 	}
+
+	/**
+	 * @Then user :user should have received an email from user :sender for sharing project space :spaceName and body of email containing
+	 *
+	 * @param string $user
+	 * @param string $sender
+	 * @param string $spaceName
+	 * @param PyStringNode $content
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function userShouldHaveReceivedAnEmailFromUserForSharingProjectSpaceAndBodyOfEmailContaining(string $user, string $sender, string $spaceName, PyStringNode $content):void {
+		$rawExpectedEmailBodyContent = \str_replace("\r\n", "\n", $content->getRaw());
+		$this->featureContext->setResponse(
+			GraphHelper::getMySpaces(
+				$this->featureContext->getBaseUrl(),
+				$user,
+				$this->featureContext->getPasswordForUser($user)
+			)
+		);
+		$expectedEmailBodyContent = $this->featureContext->substituteInLineCodes(
+			$rawExpectedEmailBodyContent,
+			$sender,
+			[],
+			[
+				[
+					"code" => "%space_id%",
+					"function" =>
+						[$this->spacesContext, "getSpaceIdByNameFromResponse"],
+					"parameter" => [$spaceName]
+				],
+			],
+			null,
+			null
+		);
+		$address = $this->featureContext->getEmailAddressForUser($user);
+		$this->featureContext->pushEmailRecipientAsMailBox($address);
+		$actualEmailBodyContent = EmailHelper::getBodyOfLastEmail($address, $this->featureContext->getStepLineRef());
+		Assert::assertStringContainsString(
+			$expectedEmailBodyContent,
+			$actualEmailBodyContent,
+			"The email address '$address' should have received an email with the body containing $expectedEmailBodyContent
+			but the received email is $actualEmailBodyContent"
+		);
+	}
+
+	/**
+	 * @Then user :user should have received an email from user :spaceName with the body containing
+	 *
+	 * @param string $user
+	 * @param string $sender
+	 * @param PyStringNode $content
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function userShouldHaveReceivedAnEmailFromUserWithTheBodyContaining(string $user, string $sender, PyStringNode $content): void {
+		$expectedContent = \str_replace("\r\n", "\n", $content->getRaw());
+		$expectedContent = $this->featureContext->substituteInLineCodes(
+			$expectedContent,
+			$sender
+		);
+		$address = $this->featureContext->getEmailAddressForUser($user);
+		$this->featureContext->pushEmailRecipientAsMailBox($address);
+		$emailBody = EmailHelper::getBodyOfLastEmail($address, $this->featureContext->getStepLineRef());
+		Assert::assertStringContainsString(
+			$expectedContent,
+			$emailBody,
+			"The email address '$address' should have received an email with the body containing $expectedContent
+			but the received email is $emailBody"
+		);
```

Comment:
This could be in separate function and be reused.
E.g. `checkEmail($user, $expectedMail)`